docs/datastore/cloudsql_mysql.md

```diff
@@ -0,0 +1,182 @@
+# Setup and configure Cloud SQL for MySQL
+
+## Before you begin
+
+1. Make sure you have a Google Cloud project and billing is enabled.
+
+1. Set your `PROJECT_ID` environment variable:
+
+    ```bash
+    export PROJECT_ID=<YOUR_PROJECT_ID>
+    ```
+
+1. [Install](https://cloud.google.com/sdk/docs/install) the gcloud CLI.
+
+1. Set gcloud project:
+
+    ```bash
+    gcloud config set project $PROJECT_ID
+    ```
+
+1. Enable APIs:
+
+    ```bash
+    gcloud services enable sqladmin.googleapis.com \
+                           aiplatform.googleapis.com
+    ```
+
+1. [Install python][install-python] and set up a python [virtual environment][venv].
+
+1. Make sure you have python version 3.11+ installed.
+
+    ```bash
+    python -V
+    ```
+
+1. Download and install [mysql-client cli (`mysql`)][install-mysql].
+
+1. Install the [Cloud SQL Auth Proxy client][install-cloudsql-proxy].
+
+[install-python]: https://cloud.google.com/python/docs/setup#installing_python
+[venv]: https://cloud.google.com/python/docs/setup#installing_and_using_virtualenv
+[install-mysql]: https://dev.mysql.com/doc/mysql-installation-excerpt/8.0/en/
+[install-cloudsql-proxy]: https://cloud.google.com/sql/docs/mysql/connect-auth-proxy
+
+
+## Create a Cloud SQL for MySQL instance
+
+1. Set environment variables. For security reasons, use a different password for
+   `$DB_PASS` and note it for future use:
+
+    ```bash
+    export DB_PASS=my-cloudsql-pass
+    export DB_USER=mysql
```

Comment:
should we not make the default user `root`?

Cloud SQL Postgres and AlloyDB use `postgres` because that is the name of the default postgres user, the default user for mysql is `root`. I think for ease of use we want to use `root` otherwise we would probably have to add a step to the doc showing the creation of the `mysql` user.

@Yuan325 what do you think?

Reply:
Yeap, if there's a default user, then let's use that :)

Reply:
done